src/main/java/edu/rpi/legup/ui/proofeditorui/treeview/TreeView.java

```diff
@@ -1,756 +1,881 @@
+                                oldElement.setCasesDepended(oldElement.getCasesDepended() - 1);
+                                Point loc = ((GridCell)oldElement).getLocation();
+                                //System.out.println("("+loc.x+","+loc.y+") to "+oldElement.getCasesDepended());
+                                if (oldElement.getCasesDepended() == 0) {
+                                    //set modifiable if started modifiable
+                                    boolean modifiable = tree.getRootNode().getBoard().getPuzzleElement(oldElement).isModifiable();
+
+                                    //unmodifiable if already modified
+                                    TreeNode modNode = ancestor.getParent().getParents().get(0);
+                                    while(modNode.getParent()!=null){
+                                        Board modBoard = modNode.getParent().getBoard();
+                                        if(modBoard.getModifiedData().contains(modBoard.getPuzzleElement(oldElement))){
+                                            modifiable = false;
+                                            break;
+                                        }
+                                        modNode = modNode.getParent().getParents().get(0);
+                                    }
+
+                                    oldElement.setModifiable(modifiable);
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+          
+            transView.getParentViews().forEach(n -> n.removeChildrenView(transView));
+            removeTreeTransition(trans);
+        }
+        repaint();
+    }
+
+    /**
+     * Called when the tree selection was changed
+     *
+     * @param selection tree selection that was changed
+     */
+    @Override
+    public void onTreeSelectionChanged(TreeViewSelection selection) {
+        this.selection.getSelectedViews().forEach(v -> v.setSelected(false));
+        selection.getSelectedViews().forEach(v -> v.setSelected(true));
+        this.selection = selection;
+        repaint();
+    }
+
+    /**
+     * Called when the model has finished updating the tree.
+     */
+    @Override
+    public void onUpdateTree() {
+        repaint();
+    }
+
+    /**
+     * Gets the TreeElementView by the corresponding TreeElement associated with it
+     *
+     * @param element TreeElement of the view
+     * @return TreeElementView of the TreeElement associated with it
+     */
+    public TreeElementView getElementView(TreeElement element) {
+        return viewMap.get(element);
+    }
+
+    private void removeTreeNode(TreeNode node) {
+        viewMap.remove(node);
+        List<TreeTransition> children = node.getChildren();
+
+        //if child is a case rule, unlock ancestor elements
+        if(!children.isEmpty()){
```

Comment:
Keep spacing consistent to the format of the existing code. There should be a space between your `if`/`for` keywords, the clause, and the opening brace.

Reply:
added for these and while loops